### ISSUE TITLE
Uncaught CancellationException (Operation was cancelled)

### ISSUE THREAD

**Issue by davidfrankl** [simolus3/moor#2180]
**After upgrading from 2.2.0 to 2.3.0, many uncaught exceptions of type `CancellationException` are thrown**

The application is set up like so:
* Background DB isolate
* Main isolate streams a select query from the table
* Background sync isolate inserts many records into the table

Previously, this worked without any exceptions. Upgrading to 2.3.0, there are many uncaught exceptions in the DB isolate with the following stack trace:
```
#0      Eval ()
#1      checkIfCancelled (package:drift/src/runtime/cancellation_zone.dart:91:5)
#2      ServerImplementation._runQuery (package:drift/src/remote/server_impl.dart:126:5)
<asynchronous suspension>
#3      DriftCommunication.setRequestHandler.<anonymous closure>.<anonymous closure> (package:drift/src/remote/communication.dart:156:13)
<asynchronous suspension>
```

Ignoring the unhandled exceptions, the behavior of the application appears to be correct.

> what you would have expected to happen

I would expect this exception to be handled somewhere in the call stack

**Comment by simolus3** [comment]
Thanks for the report! It sounds like a bug in drift, because this really should be handled somewhere in the stack. I tried to simplify that logic in 8c8c54e9922b201160d7966ac5e570d8e4de6a29, but I couldn't reproduce this issue to begin with. It probably happens when lots of queries get cancelled due to updates, but I was unable to hit that case in a test doing just that.
If you have something that reproduces this and that you could share with me, it would help a lot!

**Comment by davidfrankl** [comment]
Sorry, I still haven't had a chance to put together a small reproducible example. I'll try to pull the logic out as soon as I get a chance.

**Comment by davidfrankl** [comment]
OK, here's the repro: https://github.com/davidfrankl/drift_issue_2180: 
- start the DB on a background isolate
- start a "sync" isolate that repeatedly inserts a single row into the table.
- on the main isolate, listen to the stream, and display the number of records

The number of records loaded before first the `CancellationException` occurs varies, but in my experience, less than 15.

**Comment by simolus3** [comment]
Thank you for the helpful repro! Seems to be fixed in 8a4dea57e7d95f7584429612390197000c44b372 - could you check if that also fixes the issue in your main project by adding this to your pubspec?

```yaml
dependency_overrides:
  drift:
    git:
      url: https://github.com/simolus3/drift.git
      ref: develop
      path: drift
```

 (Unfortunately I don't really know how to test this - it's not actually an uncaught exception, setting `errorsAreFatal` on the isolate does not stop it. It just seems like the devtools need a try/catch to determine whether the async exception is uncaught?)

**Comment by davidfrankl** [comment]
It works great, thanks @simolus3!
